src/diagrams/gantt/ganttDb.js

```diff
@@ -77,23 +97,25 @@ const getStartDate = function (prevTime, dateFormat, str) {
   }
 
   // Check for actual date set
-  if (moment(str, dateFormat.trim(), true).isValid()) {
-    return moment(str, dateFormat.trim(), true).toDate()
+  let mDate = moment(str, dateFormat.trim(), true)
+  if (mDate.isValid()) {
+    return getNextValidDate(mDate, dateFormat, excludes)
```

Comment:
If an explicit enddate is set, I'm not sure whether we should change the enddate based on exclusion of weekends or other dates. You could have an explicit enddate in the weekend, because of particular reasons for that task.
The bigger question is, should enddates only be adjusted based on exclusions, when the tasks defines a duration instead of a specific enddate? I would think so.